mmcls/models/losses/eval_metrics.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import torch
+
+
+def calculate_confusion_matrix_torch(pred, target):
+    _, pred_label = pred.topk(1, dim=1)
+    num_classes = pred.size(1)
+    pred_label = pred_label.view(-1)
+    target_label = target.view(-1)
+    assert len(pred_label) == len(target_label)
+    confusion_matrix = torch.zeros(num_classes, num_classes)
+    with torch.no_grad():
+        confusion_matrix[target_label.long(), pred_label.long()] += 1
+    return confusion_matrix
+
+
+def calculate_confusion_matrix_numpy(pred, target):
+    pred_label = pred.argsort(axis=1)[:, -1]
+    num_classes = pred.shape[1]
+    pred_label = pred_label.reshape(-1)
+    target_label = target.reshape(-1)
+    assert len(pred_label) == len(target_label)
+    confusion_matrix = torch.zeros(num_classes, num_classes)
+    with torch.no_grad():
+        confusion_matrix[target_label, pred_label] += 1
+    return confusion_matrix
+
+
+def precision(pred, target):
+    """Calculate macro-averaged precision according to the prediction and target
+
+    Args:
+        pred (torch.Tensor | np.array): The model prediction.
+        target (torch.Tensor | np.array): The target of each prediction.
+
+    Returns:
+        float: The function will return a single float as precision.
+    """
+    if isinstance(pred, torch.Tensor) and isinstance(target, torch.Tensor):
+        confusion_matrix = calculate_confusion_matrix_torch(pred, target)
+    elif isinstance(pred, np.ndarray) and isinstance(target, np.ndarray):
+        confusion_matrix = calculate_confusion_matrix_numpy(pred, target)
```

Comment:
I suggest we simply use one `calculate_confusion_matrix`, process different types inside the function, and convert one to another.